src/AutorestSwift/View Models/RequestViewModel.swift

```diff
@@ -50,16 +49,24 @@ struct RequestViewModel {
         self.mediaTypes = httpWithBodyRequest?.mediaTypes ?? []
         self.knownMediaType = httpWithBodyRequest?.knownMediaType.rawValue ?? ""
 
-        // check if the request body is from signature parameter. If yes, store the object type
-        // and add the request signature parameter to operation parameters
-        var params = [ParameterViewModel]()
-        for param in request.signatureParameters ?? [] {
-            params.append(ParameterViewModel(from: param))
+        // TODO: only support the first signature parameter in Reqest now
+        let firstSignatureParam = request.signatureParameters?.first
+        self.objectType = firstSignatureParam?.schema.name
+        if isRequired(param: firstSignatureParam) {
+            self.objectName = firstSignatureParam?.name
+        } else {
+            self.objectName = "options?.\(firstSignatureParam?.name ?? "")"
         }
-        self.params = params
-
-        self.objectType = request.signatureParameters?.first?.schema.name
-        self.objectName = request.signatureParameters?.first?.name
         self.hasBody = objectType != nil
     }
 }
+
+private func isRequired(param: Parameter?) -> Bool {
```

Comment:
I'm not sure "isRequired" is the right phrase. In rare cases, you can have URI paramters that are optional. Maybe `belongsInSignature(...)`?

Reply:
Renamed.